addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,382 @@
+        """
+        return self._device_type
+
+    @classmethod
+    def supported(cls, device):
+        """
+        On specific driver override this method to check if device is supported or not
+        return True or False
+        """
+        pass
 
-    def value(self):
-        return self.value
+    def get_message(self):
+        return ''
 
     def action(self, action):
         pass
 
-    def get_connection(self):
-        return 'bluetooth'
+    def disconnect(self):
+        del iot_devices[self.device_identifier]
+
+
+#----------------------------------------------------------
+# Device manager
+#----------------------------------------------------------
+
+class EventManager(object):
+    def __init__(self):
+        self.sessions = {}
+
+    def _delete_expired_sessions(self, max_time=70):
+        '''
+        Clears sessions that are no longer called.
+
+        :param max_time: time a session can stay unused before being deleted
+        '''
+        now = time.time()
+        expired_sessions = [session for session in self.sessions if now - self.sessions[session]['time_request'] > max_time]
+        for session in expired_sessions:
+            del self.sessions[session]
+
+    def add_request(self, listener):
+        self.session = {
+            'session_id': listener['session_id'],
+            'devices': [],
+            'event': Event(),
+            'queue': {},
+            'time_request': time.time(),
+        }
+        self._delete_expired_sessions()
+        for device in listener['devices']:
+            self.session['devices'].append(device)
+        self.sessions[listener['session_id']] = self.session
+        return self.sessions[listener['session_id']]
+
+    def device_changed(self, device):
+        for session in self.sessions:
+            if device.device_identifier in self.sessions[session]['devices']:
+                self.sessions[session]['queue'] = device.data
+                self.sessions[session]['queue']['device_id'] = device.device_identifier
+                self.sessions[session]['queue']['value'] = device.value
+                self.sessions[session]['event'].set()
+
+class IoTDevice(object):
+
+    def __init__(self, dev, connection_type):
+        self.dev = dev
+        self.connection_type = connection_type
+
+event_manager = EventManager()
 
-    def connect(self):
-        pass
 
+#----------------------------------------------------------
+# Manager
+#----------------------------------------------------------
 
+class Manager(Thread):
 
+    def __init__(self):
+        super(Manager, self).__init__()
+        self.load_drivers()
 
+    def load_drivers(self): # Load drivers from IoT Box
+        path = os.path.join(home(), 'odoo/addons/hw_drivers/drivers')
+        driversList = os.listdir(path)
+        for driver in driversList:
+            path_file = path + '/' + driver
+            spec = util.spec_from_file_location(driver, path_file)
```

Comment:
clarify, what do we do here?

Reply:
1 - Creating a ModuleSpec instance based on the path to a file
2 - Create a new module based on spec
3 - Execute the module
https://docs.python.org/3/library/importlib.html